File: dice-roll-generator-javascript/src/dice_roll.js
export class DiceRoll {
    constructor() {
        this.sides = prompt("How many sides per die?");
        this.dice = prompt("Enter the number of dice to throw");
    }

    roll() {
        return Math.floor(Math.random() * this.sides) + 1;
    }

    play() {
        var sum = 0;
        var result_string = "";
        while (this.dice > 0) {
            var result = this.roll();
            sum += result;
            result_string += `${String(result)}, `;
            this.dice -= 1;
        }
        result_string += `sum is ${String(sum)}`;
        alert(`${result_string}`);
    }
}

File: dice-roll-generator-python/src/dice_roll.py
from numpy.random import randint

class DiceRoll:
    def __init__(self):
        self.sides = int(input("How many sides per dice?"))
        self.dice = int(input("How many dices to throw?"))

    def roll(self):
        return randint(1, self.sides+1)
    
    def result(self):
        sum = 0
        while self.dice>0:
            result=int(self.roll())
            print(f"{result}")
            self.dice-=1
            sum+=result
        print(f"{sum}")
